### PR TITLE
[dy] Support dbt-sqlserver

### PR DESCRIPTION
**Pull request by dy46**
# Summary

Add support for using sqlserver with dbt in Mage. Will add docs. Follows the schema here:
https://docs.getdbt.com/reference/warehouse-setups/mssql-setup

Also I used pyodbc instead of pymssql for the mssql io file because the dbt-sqlserver package uses pyodbc. Also, dbt-sqlserver doesn't technically support dbt-core 1.4.0 yet, but it seems to work based on my testing.

# Tests

tested locally.

cc: @wangxiaoyou1993

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,6 +3,13 @@ LABEL description="Deploy Mage on ECS"
 ARG PIP=pip3
 USER root
 
+# Download ODBC headers for pyodbc
+RUN curl https://packages.microsoft.com/keys/microsoft.asc | apt-key add -
+RUN curl https://packages.microsoft.com/config/debian/11/prod.list > /etc/apt/sources.list.d/mssql-release.list
+RUN apt-get update
```
Comment (wangxiaoyou1993) [inline Dockerfile:9]: there're duplicate code on `apt update`

File: mage_ai/io/mssql.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+class MSSQL(BaseSQL):
+    def __init__(
+        self,
+        database: str,
+        host: str,
+        password: str,
+        user: str,
+        schema: str = None,
+        port: int = 1433,
+        charset='utf8',
```
Comment (wangxiaoyou1993) [inline mage_ai/io/mssql.py:20]: nit: type

File: mage_ai/io/mssql.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            schema=config[ConfigKey.MSSQL_SCHEMA],
+            driver=config[ConfigKey.MSSQL_DRIVER],
+            host=config[ConfigKey.MSSQL_HOST],
+            password=config[ConfigKey.MSSQL_PASSWORD],
+            port=config[ConfigKey.MSSQL_PORT],
+            user=config[ConfigKey.MSSQL_USER],
+        )
+
+    def open(self) -> None:
+        with self.printer.print_msg('Opening connection to MySQL database'):
```
Comment (wangxiaoyou1993) [inline mage_ai/io/mssql.py:54]: nit: MSSQL

File: mage_ai/io/mssql.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        sql = f'INSERT INTO {full_table_name} VALUES ({values_placeholder})'
+        cursor.executemany(sql, values)
+
+    def get_type(self, column: Series, dtype: str) -> str:
+        if dtype in (
+            PandasTypes.MIXED,
+            PandasTypes.UNKNOWN_ARRAY,
+            PandasTypes.COMPLEX,
+        ):
+            raise BadConversionError(
```
Comment (wangxiaoyou1993) [inline mage_ai/io/mssql.py:100]: Can we use TEXT for this case?

Comment (dy46) [reply mage_ai/io/mssql.py:100]: i can try

File: Dockerfile (hunk 1 above)
Comment (dy46) [reply Dockerfile:9]: i removed the other apt update

Comment (wangxiaoyou1993) [reply Dockerfile:9]: Could you use `apt -y update` here

Comment (dy46) [reply Dockerfile:9]: oops ok let me try that

File: Dockerfile [hunk 5]
```diff
@@ -3,8 +3,15 @@ LABEL description="Deploy Mage on ECS"
...
+RUN ACCEPT_EULA=Y apt-get install -y msodbcsql18
+RUN apt install unixodbc-dev
```
Comment (wangxiaoyou1993) [inline Dockerfile:11]: Add `-y` here as well

**Pull request merged**